File: frontend/src/types/order.ts
import type { Meal } from './menu';

export interface OrderItem {
    meal: Meal;
    quantity: number;
    price: number; // in cents
}

export interface Order {
    id: number;
    user_id: number;
    week_id: number;
    status: string;
    total_price: number; // in cents
    delivery_date: string;
    items: OrderItem[];
    created_at: string;
}

export interface CheckoutRequest {
    delivery_date: string;
}

File: backend/internal/models/order.go
package models

import "time"

type Order struct {
	ID           int         `json:"id"`
	UserID       int         `json:"user_id"`
	WeekID       int         `json:"week_id"`
	Status       string      `json:"status"`
	TotalPrice   int         `json:"total_price"`
	DeliveryDate time.Time   `json:"delivery_date"`
	Items        []OrderItem `json:"items"`
	CreatedAt    time.Time   `json:"created_at"`
}

type OrderItem struct {
	OrderID  int  `json:"-"`
	MealID   int  `json:"-"`
	Meal     Meal `json:"meal"`
	Quantity int  `json:"quantity"`
	Price    int  `json:"price"` // Price at time of order (in cents)
}

type CheckoutRequest struct {
	DeliveryDate string `json:"delivery_date" binding:"required"`
}
